### DISCUSSION TITLE
Theme only applies partly

### DISCUSSION THREAD

**Discussion by Ileca** [tsujan/Kvantum#775] [Q&A] [answered]
Hello. Posting here as I saw that all issues where transferred here.

Ubuntu 21.10
Gnome 40.4.0
Wayland

I tried to apply https://github.com/vinceliuice/Graphite-kde-theme to my environment through Kvantum. It has a Kvantum theme and even recommend using Kvantum. However, it is only partially applied as you can see on my screenshot:
![Screenshot from 2022-03-21 23-06-38](https://user-images.githubusercontent.com/17970230/159371511-457ff1ae-b713-4ce9-8c01-755d3860b40d.png)

[Here is a screenshot of what it is supposed to look like, minus the colors (this is not the Nord version).](https://raw.githubusercontent.com/vinceliuice/Graphite-kde-theme/main/Graphite-dark.png)

It lacks white borders, which is the core of this theme. Possibly no round corners at the bottom (app related? as I don't see round corners on the screenshot's Dolphin? However, even Kvantum doesn't have round corners at the bottom). Possibly a little transparency problem with the Name/Size bar above files but that might be a problem with the theme itself.

**Comment by tsujan** [comment, marked as answer]
I checked the theme. It was OK.

> It lacks white borders, which is the core of this theme.

Window borders (= window decoration) aren't related to Kvantum but to the window manager (e.g., KWin in KDE).

**Reply by Ileca** [reply]
Meaning you have bottom round corners and no transparency for the column names on Dolphin? Or are bottom corners also window decorations, or just upper corners? And everything else is the same as seen on my screenshot?

>Window borders (= window decoration) aren't related to Kvantum but to the window manager (e.g., KWin in KDE).

I see. Don't laugh but I applied the GTK version of this theme too.

**Reply by tsujan** [reply]
A Kvantum theme cannot be applied "partially". What you see in Graphite screenshots are combinations of a Kvantum theme (for everything inside windows) + an Aurorae theme (for KDE's WM) + a Plasma theme (for KDE's Plasma).

Since you use Gnome, Gnome's WM is responsible for the borders of windows. Yes, all of them are parts of the decoration.

EDIT: WM = Window Manager = a program that controls decorations and also how windows behave.

**Reply by Ileca** [reply]
Thanks for taking the time to explain that. I started Linux 1-2 month ago and I have a hard time figuring out all those different little pieces. I can't find a good doc explaining all of that. Ricing your Linux is hard... (yes, I recently learnt this word)
More question, please:
* Why do you need both a Kvantum theme and a Plasma theme? Aren't Plasma themes supposed to handle the inside of windows?
* Instead of Plasma themes, shouldn't you say Qt themes? Aren't Qt and GTK what matters here?
* If the WM is handling windows, why does my GTK apps have window decorations, but not my Qt apps? This is my main problem.
* On what the status bar, dock and contextual menus are relying on to be decorated?

**Reply by tsujan** [reply]
> Why do you need both a Kvantum theme and a Plasma theme?

You'll need a Plasma theme only if you use KDE. "Plasma" is the name KDE developers gave to their desktop widgets (their panel and so-called "Plasmoids") since KDE4. Gnome doesn't have Plasma; instead, it has what they call "Gnome Shell".

> Instead of Plasma themes, shouldn't you say Qt themes? Aren't Qt and GTK what matters here?

No. Gnome has two different themes: GTK theme and gnome-shell theme. Almost in the same sense, KDE has a Plasma theme, that's totally different from a Qt theme.

> If the WM is handling windows, why does my GTK apps have window decorations, but not my Qt apps?

Gnome has its own way of handling windows: it has CSD (= client-side decoration). CSD is only for GTK apps. When you run a Qt app under Gnome, its window gets a title-bar different from that of CSD. Blame that on Gnome ;)

KDE doesn't have CSD. Its WM (KWin) supports various decorations. By default, it uses Breeze, but there are other decorations like Aurorae, BreezeEnhanced, etc. The user can install any of them but he/she can use only one in each session.

In Linux, you encounter concepts that don't exist in MS Windows because the latter has only one of each. For example, Windows users are usually not very familiar with terms like "window manager", "file manager", "desktop display manager", or "DE" (= Desktop Environment). Although Windows has them, since there's no variety, the user accepts them as inseparable parts of Windows. In contrast, you could change almost everything in Linux.

Last but not least, KDE ≠ Qt, and Gnome ≠ GTK. Gnome and KDE are only two DEs among many. There are others like XFCE, LXQt, Cinnamon, Enlightenment,..., each one with its own WM, panel, file manager, etc. Enlightenment even has its own toolkit: it's called "EFL", which is totally different from GTK and Qt.

Why is there so much variety in Linux? Well, I don't have a definite answer, but I wouldn't use Linux if there wasn't ;)

**Reply by Ileca** [reply]
> You'll need a Plasma theme only if you use KDE. "Plasma" is the name KDE developers gave to their desktop widgets (their panel and so-called "Plasmoids") since KDE4. Gnome doesn't have Plasma; instead, it has what they call "Gnome Shell".

Ok, so if I had not Kvantum, which is agnostic, to decorate the inside of the apps, I would have to apply GTK or Qt themes depending on what the app use, if the DE allows it (and the reason why I need to install Kvantum to decorate Qt apps in Gnome)? It also handles contextual menus from the apps themselves.
And to decorate the status bar, context menus (from the DE, like right clicking on the desktop) and widgets, themes for Plasma and Gnome Shell?
That's why you have so many themes in the KDE Graphite theme:
* Aurorae Themes < WM stuff, tho I thought: isn't Aurorea a window decoration theme for KWin? In that case, if you made a Graphite theme, should you say it's a "Graphite Theme for KWin" or "KWin Theme"? But then I read Aurorae is a "theme engine"... does that mean you need to use a certain API to theme the WM? And there are multiple ones like BreezeEnhanced you made? Why isn't there a single way to decorate WM stuff? This sounds like complicating stuff for nothing, here. Or is it because some of those API have features some others don't have like Kvantum which uses SVG?
* Kvantum Themes < inside Qt apps (a Qt theme would have been the same (but unable to use SVG))
* Plasma Color Schemes < dunno what's the difference between Color Schemes/Desktop Themes/Global Themes but all of that if for KDE specific desktop stuff
* Plasma Desktop Themes
* Plasma Global Themes

> When you run a Qt app under Gnome, its window gets a title-bar different from that of CSD. Blame that on Gnome ;)

Honestly, that boggles my mind. That sounds so stupid. It's like they are deliberately discriminating against Qt apps on their desktop. Btw, this is not totally exact because if I change the theme, the handle of Qt apps is matching the theme's colors and upper corners' radius. It is changed, but I don't have the white outline all around the app that I want. Meaning a GTK theme partially decorates Qt apps? Or is the Graphite theme handling this non CSD stuff but it is bugged and only partially support whatever Gnome uses for Qt apps?

What's a Cairo/Murrine engine? See, it says that it is used "to draw things to the screen or to generate output for printing" but that's not helping AT ALL... >.>
It is a dependency of the [GTK theme](https://github.com/vinceliuice/Graphite-gtk-theme) so is it maybe a theme engine like Aurorae/BreezeEnhanced, but for Gnome's WM? Don't tell me GTK have multiple engines too... OTL

To decorate my Gnome status bar's menus, I need a Gnome shell theme... Despite those menus applying the Graphite GTK theme on the screenshots, it is not applied on my Gnome. Bug? I thought maybe they are showing GTK apps inside KDE (using Plasma Theme) but it's the Gnome status bar, meaning the theme should handle that...

> Why is there so much variety in Linux? Well, I don't have a definite answer, but I wouldn't use Linux if there wasn't ;)

Yeah, Linux's variety and modularity are both its strength and weakness. On one side, you can have a full custom experience, but on the other side, there is plenty of problems with interoperability. For the problem at hand, the worst was that I had no idea who I was supposed to bug with my problems. In the end, I did well not bothering the theme's author like they had nothing to do with that.
I am starting to understand that it's probably better if I stick to a subset of apps that integrate well with each others.

**Reply by tsujan** [reply]
> That sounds so stupid. It's like they are deliberately discriminating against Qt apps on their desktop.

Stupid or not, Gnome is known for not caring about anything other than Gnome.

> Btw, this is not totally exact because if I change the theme, the handle of Qt apps is matching the theme's colors

If a Kvantum theme has a dark variant — as Graphite does — Gnome's title-bar will become dark with it. Kvantum includes the needed code. The code works for now but, in a future version of Gnome, it might not. If it stops working, I'll have to remove it and forget about Gnome altogether. Gnome isn't Qt-friendly and I have no reason to tolerate that beyond a point.

> it is not applied on my Gnome. Bug?

It isn't a bug. The Graphite collection is *for KDE*, not for Gnome. Only its Kvantum theme can be used under Gnome because only Kvantum is DE-agnostic.

If you like Gnome, try to use the apps that are designed for it. Under Gnome, you could use a few Qt apps but Gnome isn't Qt-friendly. A KDE app like Dolphin can give you a headache under Gnome. Purely Qt-based apps may work without problem though (as I said, KDE ≠ Qt).

If you like Qt apps but want to have a few GTK apps (like Firefox, Inkscape, GIMP,...), KDE might be a better choice.

**Reply by Ileca** [reply]
> The Graphite collection is for KDE, not for Gnome.

The Graphite theme has a KDE theme, and a GTK theme I linked it my last post.
[https://github.com/vinceliuice/Graphite-kde-theme](https://github.com/vinceliuice/Graphite-kde-theme)
[https://github.com/vinceliuice/Graphite-gtk-theme](https://github.com/vinceliuice/Graphite-gtk-theme)

The screenshots of the GTK theme use Gnome.

> A KDE app like Dolphin can give you a headache under Gnome.

You have no idea!

> KDE might be a better choice.

I was thinking about doing that but as a noob, I chose the Ubuntu noob experience the time to get more experience. A tiling WM like Sway actually is something I would like to try. More headaches for me.

**Comment by tsujan** [comment]
> Posting here as I saw that all issues where transferred here.

Issues aren't transferred to Discussions. Since your problem wasn't about an issue in Kvantum, you did the right thing to choose Discussions → Q&A :)
